frontend/src/app/services/scenario.ts

```diff
@@ -73,30 +73,36 @@ const czechiaGame = new Scenario({
 // reproduce mitigation actions of Czech Republic up to December 2020
 const czechiaReal = new Scenario({
   rampUpStartDate: '2020-02-25',
-  rampUpEndDate: '2020-12-18',
+  rampUpEndDate: '2021-01-09',
   endDate: '2021-07-01',
 });
 
 // First wave
-czechiaReal.addMitigationAction(['rrr', true], '2020-03-14');
+czechiaReal.addMitigationAction(['rrr', true], '2020-03-14', '2020-06-01');
 czechiaReal.addMitigationAction(['events', 100], '2020-03-10');
 czechiaReal.addMitigationAction(['events', 10], '2020-03-24');
-czechiaReal.addMitigationAction(['events', 100], '2020-05-12');
-czechiaReal.addMitigationAction(['events', 1000], '2020-05-26');
+czechiaReal.addMitigationAction(['events', 100], '2020-05-11');
+czechiaReal.addMitigationAction(['events', 1000], '2020-05-25');
 czechiaReal.addMitigationAction(['businesses', 'most'], '2020-03-14');
-czechiaReal.addMitigationAction(['businesses', 'some'], '2020-05-12', '2020-07-01');
+czechiaReal.addMitigationAction(['businesses', 'some'], '2020-05-11', '2020-07-01');
 czechiaReal.addMitigationAction(['schools', 'all'], '2020-03-13', '2020-08-31');
-czechiaReal.addMitigationAction(['stayHome', true], '2020-03-16', '2020-04-24');
-czechiaReal.addMitigationAction(['bordersClosed', true], '2020-03-16', '2020-04-24'); // TODO find end date
+czechiaReal.addMitigationAction(['stayHome', true], '2020-03-16', '2020-04-25');
```

Comment:
Both dates are inclusive (the mitigation is active in them). Please, just confirm you are aware of that.
Btw, you added a condition `!(id in oldAction.mitigations` to the method earlier. I do not think is is necessary/predictive - as we are adding mitigations to the scenario, later rules should override earlier ones in case they overlap. In either case definitions should be done chronologically, if 2 (non-default) mitigations in one variable are done in a row, no end date is needed for the 1st one.